### PR TITLE
BOOKKEEPER-937 Upgrade protobuf to 2.6

### PR DESCRIPTION
**Pull request by dlg99**
protobuf/protoc 2.4 cannot be installed with brew on mac and building it on mac always result is build errors hence leaves an option of switching to linux to run protoc.
I upgraded to 2.6 instead. It is compatible with 2.4 on the wire and shaded so should not create any problems. All tests passed.
Please ignore changes in java files in attached patch during review; these are auto-generated.

protobuf 2.4 in compat* dependencies is shaded now to not conflict with 2.6.

### CODE REVIEW

File: bookkeeper-server/src/test/proto/TestDataFormats.proto [hunk 1]
```diff
@@ -30,5 +30,5 @@ enum AuthMessageType {
  *
  */
 extend AuthMessage {
-    required AuthMessageType messageType = 1000;
+    optional AuthMessageType messageType = 1000;
```
Comment (eolivelli) [inline bookkeeper-server/src/test/proto/TestDataFormats.proto]: Why did you change this to "optional" ? anyway it is used only in tests there is no matter of compatibility with previous versions

Comment (dlg99) [inline bookkeeper-server/src/test/proto/TestDataFormats.proto]: @eolivelli protobuf 2.6 no longer allows ading required fields to extended messages. Protoc errors out.
Please see this conversation for details: https://groups.google.com/forum/#!topic/protobuf/EyR1EdDL5cs 
"Extensions, by its definition, is something optionally attached to a message and it doesn't make sense for it to be required, and technically there is no way to enforce this requirement because a binary can be compiled without the extension definition in which case there is no way to know whether the extension is required or optional.

And the change in 2.6.0 is actually a bug fix because in previous versions the required label is simply ignored and the extension is treated as optional any way."

Comment (eolivelli) [inline bookkeeper-server/src/test/proto/TestDataFormats.proto]: Thank you for your explanation

**Comment by dlg99** [comment]
I looked at the jenkins failure for the pull request and it does not look related to the changes. 
It looks like BOOKKEEPER-876 
I cannot repro it locally.

**Comment by eolivelli** [comment]
I'm going to test your patch as soon as possible on my laptop. It appears to be a transient failure

**Comment by eolivelli** [comment]
I checked out out the patch and no test is failing on my machine
LGTM
 +1

**Comment by jiazhai** [comment]
LGTM
+1

**Comment by sijie** [comment]
+1 from me as well. will commit this soon.

**Comment by sijie** [comment]
@dlg99 can you rebase the pull request? so I can merge it. thank you so much.

**Comment by dlg99** [comment]
@sijie Rebased. Tests passed locally.
